json-fleece-core/src/Fleece/Core/Schemas.hs

```diff
@@ -251,12 +256,36 @@ list itemSchema =
     V.toList
     (array itemSchema)
 
-map :: (Fleece schema, Typeable a) => schema a -> schema (Map.Map T.Text a)
-map innerSchema =
+jsonMap :: (Fleece schema, Typeable a) => schema a -> schema (Map.Map T.Text a)
+jsonMap innerSchema =
   object $
     constructor id
       #* additionalFields id innerSchema
 
+mapFrom ::
+  (Fleece schema, Ord a, Typeable a, Typeable b) =>
+  (b -> b -> b) ->
+  schema (a, b) ->
+  schema (Map.Map a b)
+mapFrom combiner assocSchema =
+  transform Map.toList (Map.fromListWith combiner) (list assocSchema)
+
+strictMapFrom ::
+  (Fleece schema, Ord a, Typeable a, Typeable b) =>
+  (b -> b -> b) ->
+  schema (a, b) ->
+  schema (StrictMap.Map a b)
+strictMapFrom combiner assocSchema =
+  transform StrictMap.toList (StrictMap.fromListWith combiner) (list assocSchema)
+
+intMapFrom ::
+  (Fleece schema, Typeable a) =>
+  (a -> a -> a) ->
+  schema (Int, a) ->
+  schema (IntMap.IntMap a)
+intMapFrom combiner assocSchema =
+  transform IntMap.toList (IntMap.fromListWith combiner) (list assocSchema)
```

Comment:
I think we need to discuss the naming of the these functions. I fully expected them to be json objects but in fact they are lists in the underlying json